lndr-backend/src/Lndr/Handler/Credit.hs

```diff
@@ -135,12 +139,12 @@ finalizeCredit bilateralCredit = do
                 commit conn
 
 
-createPendingRecord :: Pool Connection -> CreditRecord -> LndrHandler ()
-createPendingRecord pool signedRecord = do
+createPendingRecord :: Int -> Pool Connection -> CreditRecord -> LndrHandler ()
+createPendingRecord recordNum pool signedRecord = do
     -- TODO can I consolidate these two checks into one?
     -- check if a pending transaction already exists between the two users
     existingPending <- liftIO . withResource pool $ Db.lookupPendingByAddresses (creditor signedRecord) (debtor signedRecord)
-    unless (null existingPending) $
+    unless ((recordNum > 0) || (null existingPending)) $
```

Comment:
📝This conditional is a bit difficult to read, because there's a lot going on here.

 1. Compound predicates are more difficult to read than simple predicates (e.g: `if (A || B)` vs `if (A)`).

 2. `unless` is a negative operator, so it inverts the logic of the predicate (effectively adding an implicit `NOT` to the clause). The reader, then, has to perform this inversion mentally to understand the effect.

 3. The first part of the predicate is a comparator with the expected lesser value on the right-hand side. *greater-than* is slightly more difficult to read than *less-than* for native readers of left-to-right languages because of the natural tendency to expect values to be increasing from left to right. *greater-than* causes the flow to temporarily reverse, resulting in some disruption to readability.

To address these concerns:

 1. Transform the logic of this predicate such that it uses an `&&` rather than a `||` using [DeMorgan's Laws](https://en.wikipedia.org/wiki/De_Morgan%27s_laws). (`!(P && Q) <=> !P || !Q`)
 2. Split the compound predicate into a nested conditional such that each line is easier to read.
 3. Adjust the resulting `recordNum > 0` operation, if necessary, to use a *less-than* rather than a *greater-than*.

This gives us:

```
# Before
unless ((recordNum > 0 || (null existingPending)) $

# Transform to if-statement
if (!((recordNum > 0 || (null existingPending))) $

# Apply DeMorgan's transformation
if (!!(recordNum <= 0) && !(null existingPending)) $

# Double-negation cancels out
if (recordNum <= 0 && !(null existingPending)) $

# Use nested conditional instead of &&
if (recordNum <= 0) $
  if (!(null existingPending)) $
    throwError (err400 …)
```

❤️ Readability! The #1 rule for readability: each line should express a single, complete thought. Or, with respect to conditionals, a single logical branch.